kentikapi/internal/api_resources/sites_test.go

```diff
@@ -0,0 +1,209 @@
+	transport := &api_connection.StubTransport{ResponseBody: getResponsePayload}
+	sitesAPI := api_resources.NewSitesAPI(transport)
+	siteID := models.ID(42)
+
+	// act
+	site, err := sitesAPI.Get(nil, siteID)
+
+	// assert request properly formed
+	assert := assert.New(t)
+	require := require.New(t)
+
+	require.NoError(err)
+	assert.Zero(transport.RequestBody)
+
+	// and response properly parsed
+	assert.Equal(models.ID(42), site.ID)
+	assert.Equal("apitest-site-1", site.SiteName)
+	assert.Equal(54.349276, *site.Latitude)
+	assert.Equal(18.659577, *site.Longitude)
+	assert.Equal(models.ID(3250), site.CompanyID)
+}
+
+func TestGetAllSites(t *testing.T) {
+	// arrange
+	getResponsePayload := `
+	{
+		"sites": [
+			{
+				"id": 7758,
+				"site_name": "AWS us-east-1",
+				"lat": null,
+				"lon": null,
+				"company_id": 74333
+			},
+			{
+				"id": 8483,
+				"site_name": "marina gdańsk",
+				"lat": 54.348972,
+				"lon": 18.659791,
+				"company_id": 74333
+			},
+			{
+				"id": 8592,
+				"site_name": "mysite",
+				"lat": 1,
+				"lon": 2,
+				"company_id": 74333
+			}
+		]
+	}`
+	transport := &api_connection.StubTransport{ResponseBody: getResponsePayload}
+	sitesAPI := api_resources.NewSitesAPI(transport)
+
+	// act
+	sites, err := sitesAPI.GetAll(nil)
+
+	// assert request properly formed
+	assert := assert.New(t)
+	require := require.New(t)
+
+	require.NoError(err)
+	assert.Zero(transport.RequestBody)
+
+	// and response properly parsed
+	require.Equal(3, len(sites))
+	// site 0
+	site := sites[0]
+	assert.Equal(models.ID(7758), site.ID)
+	assert.Equal("AWS us-east-1", site.SiteName)
+	assert.Nil(site.Latitude)
+	assert.Nil(site.Longitude)
+	assert.Equal(models.ID(74333), site.CompanyID)
+	// site 1
+	site = sites[1]
+	assert.Equal(models.ID(8483), site.ID)
+	assert.Equal("marina gdańsk", site.SiteName)
+	assert.Equal(54.348972, *site.Latitude)
+	assert.Equal(18.659791, *site.Longitude)
+	assert.Equal(models.ID(74333), site.CompanyID)
+	// site 2
+	site = sites[2]
+	assert.Equal(models.ID(8592), site.ID)
+	assert.Equal("mysite", site.SiteName)
+	assert.Equal(1.0, *site.Latitude)
+	assert.Equal(2.0, *site.Longitude)
+	assert.Equal(models.ID(74333), site.CompanyID)
+}
+
+func TestCreateSite(t *testing.T) {
+	// arrange
+	createResponsePayload := `
+	{
+		"site": {
+			"id": 42,
+			"site_name": "apitest-site-1",
+			"lat": 54.349276,
+			"lon": 18.659577,
+			"company_id": "3250"
+		}     
+	}`
+	transport := &api_connection.StubTransport{ResponseBody: createResponsePayload}
+	sitesAPI := api_resources.NewSitesAPI(transport)
+
+	// act
+	site := models.NewSite("apitest-site-1")
+	models.SetOptional(&site.Longitude, 18.659577)
+	models.SetOptional(&site.Latitude, 54.349276)
+	created, err := sitesAPI.Create(nil, *site)
+
+	// assert request properly formed
+	assert := assert.New(t)
+	require := require.New(t)
+
+	require.NoError(err)
+	payload := utils.NewJSONPayloadInspector(t, transport.RequestBody)
+	require.NotNil(payload.Get("site"))
```

Comment:
I think that we should check that the library does not send redundant data in request body, e.g.
```json
{
  "site": {
    "site_name": "foo",
    "redundant_trash": "trash",
  }
}
```

I suggest to check number of keys in JSON objects.

Reply:
We can extend the json payload inspector with such functionality in a separate task

Reply:
Ok, so please create such task

Reply:
Task created: https://clkentik.atlassian.net/browse/KNTK-183